src/NCronJob/Registry/JobRun.cs

```diff
@@ -1,36 +1,50 @@
 
+using System.Collections.Concurrent;
+
 namespace NCronJob;
 
 internal class JobRun
 {
+    private readonly JobRun rootJob;
     private int jobExecutionCount;
+    private readonly Action<JobRun> progressReporter;
+    private readonly ConcurrentBag<JobRun> pendingDependents = [];
 
     private JobRun(
         JobDefinition jobDefinition,
-        object? parameter)
-    : this(null, jobDefinition, parameter)
+        object? parameter,
+        Action<JobRun> progressReporter)
+    : this(null, jobDefinition, parameter, progressReporter)
     { }
 
     private JobRun(
         JobRun? parentJob,
         JobDefinition jobDefinition,
-        object? parameter)
+        object? parameter,
+        Action<JobRun> progressReporter)
     {
         var jobRunId = Guid.NewGuid();
 
         JobRunId = jobRunId;
+        ParentJobRunId = parentJob is not null ? parentJob.JobRunId : jobRunId;
```

Comment:
That is interesting. Is there a reason not to set `ParanetJobRunId` to `null` if there isn't any parent?